### PR TITLE
Added DataframeTableDescriptor

### PR DESCRIPTION
**Pull request by prutskov**

### CODE REVIEW

**Comment by prutskov** [comment]
@fexolm review, please

File: Catalog/Catalog.cpp [hunk 1]
```diff
@@ -2093,7 +2094,8 @@ void Catalog::createTable(
     if (td.persistenceLevel == Data_Namespace::MemoryLevel::DISK_LEVEL) {
       throw std::runtime_error("Only temporary tables can be backed by foreign storage.");
     }
-    ForeignStorageInterface::prepareTable(getCurrentDB().dbId, td, cds);
+    DataframeTableDescriptor& dtd = static_cast<DataframeTableDescriptor&>(td);
```
Comment (fexolm) [inline Catalog/Catalog.cpp]: maybe df_td?

**Comment by fexolm** [comment]
LGTM for me, but I'm not well aware of that code. Let's merge and see Alex's response.

File: Catalog/Catalog.cpp (hunk 1 above)
Comment (prutskov) [reply Catalog/Catalog.cpp]: fixed

**Pull request merged**